### DISCUSSION TITLE
Only for web applications?

### DISCUSSION THREAD

**Discussion by krabbenprgr** [passwordless-lib/fido2-net-lib#573] [Q&A]
Cool package, thanks for that. 
I just wonder if it is also suitable for .Net applications that are not a web page (Windows Forms Application). Does the package also work offline?
Also, I wonder if you can do without Windows Hello, since I don't want Windows to open a window on top of my application. I'd rather I could convey the instructions to the user (for example, to put on the Fido stick) directly through my app.

Thanks in advance.

**Comment by aseigler** [comment]
On recent builds of Windows 10 and all builds of 11, the OS brokers the conversation with the authenticator through the familiar credential dialog.  In order to bypass that, you have to be running as administrator, which is a security risk of course.  There's no reason why it couldn't work offline, but you'd have to carefully consider how you built the application if you want to not use the Hello dialog.

**Comment by FrankSzendzielarz** [comment]
Second this. I am not sure looking at this library where the "native" support is. Can someone point to an example of using this in a console application rather than webauthn?
